---
title: "transcriptomics covid"
output: html_notebook
---


```{r}
library(dplyr)
library(xlsx)
library(ggplot2)
```

## list significant proteomics
```{r}
sign_prot <- read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/INF_Grant/processing/proteins_neogi_interferon_types.txt")
sign_prot <- as.vector(sign_prot$x)
H0_48 <- read.xlsx("/home/flomik/Desktop/Code-PHD/Corona_projects/INF_Grant/data/Neogi/DGE/covid-19_proteomics_UNINF_vs_H48PI_DGE_results_filt.xlsx", 1)
sign_prot <- sign_prot[sign_prot %in% H0_48$Gene_name]
```

## Filter for interferon
```{r}
antiviral <- read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/INF_Grant/data/databases/step_3/antiviral_mechanism_IFN.tsv")
antiviral <- gsub(".* ", "", antiviral$MoleculeName)
inf_alpha_beta <- read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/INF_Grant/data/databases/step_3/reactome_interferon_alpha_beta.tsv")
inf_alpha_beta <- gsub(".* ", "", inf_alpha_beta$MoleculeName)
inf_gamma <- read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/INF_Grant/data/databases/step_3/reactome_interferon_gamma.tsv")
inf_gamma <- gsub(".* ", "", inf_gamma$MoleculeName)
```

```{r}
inf <- unique(c(antiviral, inf_alpha_beta, inf_gamma))
```

## load dge
```{r}
dge_24 <- read.csv("/home/flomik/Desktop/Code-PHD/Corona_projects/Corona_omics/results/covid-19_transcriptomics/DGE/covid-19_transcriptomics_UNINF_vs_H24PI_DGE_results.csv")
dge_24$logFC <- ifelse(dge_24$adj.P.Val > 0.05, NA , dge_24$logFC)
dge_24 <- select(dge_24, Ensembl_ID, Gene_name, logFC)
names(dge_24)[3] <- "LFC_24"
dge_48 <- read.csv("/home/flomik/Desktop/Code-PHD/Corona_projects/Corona_omics/results/covid-19_transcriptomics/DGE/covid-19_transcriptomics_UNINF_vs_H48PI_DGE_results.csv")
dge_48$logFC <- ifelse(dge_48$adj.P.Val > 0.05, NA , dge_48$logFC)
dge_48 <- select(dge_48, Ensembl_ID, logFC)
names(dge_48)[2] <- "LFC_48"
#dge_72 <- read.csv("/home/flomik/Desktop/Code-PHD/Corona_projects/Corona_omics/results/covid-19_transcriptomics/DGE/covid-19_transcriptomics_UNINF_vs_H72PI_DGE_results.csv")
#dge_72$logFC <- ifelse(dge_72$adj.P.Val > 0.05, NA , dge_72$logFC)
#dge_72 <- select(dge_72, Ensembl_ID, logFC)
#names(dge_72)[2] <- "LFC_72"
```
```{r}
dge <- merge(dge_24, dge_48, by = "Ensembl_ID", all.x = TRUE, all.y = TRUE)
#dge <- merge(dge, dge_72, by = "Ensembl_ID", all.x = TRUE, all.y = TRUE)
```

```{r}
names(dge_24)
```


## load data
```{r}
data <-read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/Covid_India/data/transcriptomics_Coding_Count.txt")
```

```{r}
names(data)[1] <- "Ensembl_ID"
```

```{r}
head(data)
```

```{r}
annotation_info_1 <- read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/Covid_India/data/annotation_ensembl_id_hugo_id.txt", header = FALSE, stringsAsFactors = FALSE)
annotation_info_2 <- read.delim("/home/flomik/Desktop/Code-PHD/Corona_projects/Covid_India/data/annotationInfo_Uniprot_gene_name", header = FALSE, stringsAsFactors = FALSE)
```

```{r}
names(annotation_info_1) <- c("Accession", "Ensembl_ID")
names(annotation_info_2) <- c("Accession", "Gene_name")
annotationInfo <- merge(annotation_info_2, annotation_info_1, by = "Accession")
```

```{r}
names(annotationInfo)
```

```{r}
names(data)[1] <- "Ensembl_ID"
data <- merge(data, annotationInfo, by = "Ensembl_ID")
```

```{r}
data <- data[data$Gene_name %in% inf,]
data <- data[!duplicated(data$Gene_name),]
```

```{r}
#data$Ensembl_ID <- NULL
data$Accession <- NULL
rownames(data) <- data$Gene_name
data$Gene_name <- NULL
```

```{r}
names(data)
```

```{r}
data <- select(data, Ensembl_ID, contains("Uninf"), contains("24"), contains("48"))
```
```{r}
data <- merge(data, dge, by = "Ensembl_ID")
```

```{r}
length(!is.na(data$LFC_48))
```

```{r}
input <- data
```

```{r}
input$Pathway <- ifelse(input$Gene_name %in% inf_alpha_beta, "Interferon alpha/beta signaling", NA)
input$Pathway <- ifelse(input$Gene_name %in% inf_gamma, "Interferon gamma signaling", input$Pathway)
input$Pathway <- ifelse(input$Gene_name %in% antiviral, "Antiviral mechanism by IFN-stimulated genes", input$Pathway)
```

```{r}
library(circlize)
library(ComplexHeatmap)
col_fun_2 = colorRamp2(c(-1.5,  - 0.001 ,0,  0.001, 1.5), c("blue", "#b3ccff","white","#ff9999", "red"))

col_fun1 = colorRamp2(c(-2,-1,-0.5, 0,0.5,1,2), c("#7F7F00","#B2B200" ,"#E5E500","white","#BF7FBF","#993299","#590059"))

col_fun = colorRamp2(c(-2, 0, 2), c("green", "white", "red"))
```
```{r}
condition <- factor(c(rep("Mock", 3), rep("24hpi", 3), rep("48hpi", 3), rep("72hpi", 3)), levels = c("Mock", "24hpi", "48hpi", "72hpi"))
```
```{r}
c_t0 <- "#B1A78C"
c_t24 <- "#687E8A"
c_t48 <- "#41607D"
c_t72 <- "#09253B"
```

```{r}
list_network <- c("PIN1", 
                  "UBA52",
                  "MAPK3",
                  "PRKCD",
                  
                  "EIF2AK2",
                  "DDX58",
                  "EIF4A1",
                  
                  "JAK1",
                  "STAT1",
                  "PIAS1",
                  "ISG15",
                  "RPS27A",
                  "EIF4A2",
                  
                  "MX2",
                  "IFIT1",
                  
                  "PTPN1",
                  "STAT2",
                  "GBP2",
                  "IRF2",
                  "IFI35",
                  "PPM1B",
                  "IFI30",
                  "ICAM1",
                  ""
                  
                  )
```


```{r}
#input_beta <- input[input$Gene_name %in% sign_prot,]
input_beta <- input[input$Gene_name %in% list_network,]
```
```{r}
proteins <- input_beta$Gene_name
```
```{r}
names(input)
```

```{r}
input[,-c(1, 11:14)] <- t(scale(t(input[,-c(1, 11:14)])))
```
```{r}
rownames(input) <- input$Gene_name
```


```{r}
pdf("Heatmat_neogi_covid_interferon_transcriptomics.pdf", width = 10, height = 22)
ht = Heatmap(as.matrix(input[,-c(1, 11:14)]),
             col = col_fun1,
             show_row_names = TRUE,
             column_order = colnames(input)[-c(1, 11:14)],
             show_column_names = FALSE,
             column_split = condition[-c(10:13)],
             row_split = input$Pathway,
             row_title_rot = 0,
             column_title_gp = gpar(fill = c(c_t0, c_t24, c_t48)),
             border = FALSE,
             column_gap = unit(2, "mm"), 
             right_annotation = rowAnnotation(Mock_H24 = input$LFC_24,
                                              Mock_H48 = input$LFC_48,
                                              #Mock_H72 = input$LFC_72,
                                              
                                              col = list(
                                                         Mock_H24 = col_fun_2,
                                                         Mock_H48 = col_fun_2
                                                         #Mock_H72 = col_fun_2
                                                         ),  show_legend = TRUE, gp = gpar(col = "grey")
             ))


#hm = Heatmap(as.matrix(input[,c(15:17)]), col = col_fun, cluster_rows = FALSE, cluster_columns = FALSE , column_names_rot = 0, column_names_side = "top")

ht_list = ht

draw(ht_list)
dev.off()
```

```{r}
input_beta[,-c(1, 11:14)] <- t(scale(t(input_beta[,-c(1, 11:14)])))
```


```{r}
input_beta_2 <- data.frame(t(input_beta[,-c(1, 11:14)]), condition = condition[-c(10:13)])
names(input_beta_2) <- c(as.vector(proteins), "condition")
```

```{r}
data_means <- aggregate(. ~ condition, data=input_beta_2, mean)
```
```{r}
data_means$condition <- NULL
```
```{r}
names(data_means)
```
```{r}
names(input_beta)
```

```{r}
list_figures <- list()
j = 1
```

```{r}
condition
```
```{r}
for (i in 1:ncol(data_means)) {
  data <- data.frame(value = 1:9, condition = condition[-c(10:13)], replicate = rep(c(1, 2, 3), 3))
  mean_d <- data.frame(value = 1:3, condition = unique(as.vector(condition[-c(10:13)])))
  print(mean_d)
  data$value <- input_beta_2[,i]
  data$prot <- colnames(input_beta_2)[i]
  mean_d$value <- data_means[,i]
  mean_d$replicate <- 1
  mean_d$group <- c("1", "2", "3")
  data$group <- rep(c("1", "2", "3"), each = 3)
  mean_d$prot <- colnames(input_beta_2)[i]
  print(mean_d)
  print(data)
  pltbase <- ggplot()+ geom_line(data = data, aes(x=group, y=value, group = replicate), size=2, color = "#e0c4c6", alpha=0.6)+ theme_classic()+ scale_x_discrete(labels=c("1" = "Mock", "2" = "24hpi", "3" = "48hpi", "4" = "72hpi"))
  pltbase <-pltbase + coord_cartesian(xlim = c(0.95, 3.2), expand = FALSE) + theme(axis.title.x=element_blank(), axis.title.y=element_blank(), axis.text.x = element_text(angle = 45, hjust = 1, size=8, color = "black"), axis.text.y = element_text(size=10, color = "black"))
  pltbase <- pltbase + geom_line(data=mean_d, aes(x=group, y=value, group = 1), color="#983c44", size = 1.5) + ggtitle(colnames(input_beta_2)[i])+geom_point(data = mean_d, aes(x=group, y=value, group = replicate), size = 2, color = "#983c44", alpha = 1)
  
  print(pltbase)
  list_figures[[j]] = pltbase
  j <- j+1
  
}
```

```{r}
require(gridExtra)
```
```{r}
m <- matrix(1:25, nrow = 5, ncol = 5)
```

```{r}
pdf("Genes_significant_network_transcriptomics_data.pdf")
grid.arrange(grobs = list_figures, layout_matrix = m)
dev.off()
```
